Untitled Pull Request

etdevios opened the pull request:
Реализован сплеш-скрин и основной экран.

etdevios committed:
Добавлены шрифты: YS Display-Bold, YS Display-Medium и цвета: YP Background, YP Black, YP Gray, YP Green, YP Red, YP White

etdevios committed:
Создан и настроен Launch Screen

etdevios committed:
Создан первый экран приложения в IB, определены положения и размеры элементов через Auto Layout

etdevios committed:
Реализованы все состояния и переходы конечного автомата основного экрана.

etdevios committed:
fix: созданы связи в IB actions и outlets

mmbabaev commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,10 +1,173 @@
+        }
+        
+        imageView.layer.masksToBounds = true
+        imageView.layer.borderWidth = 8
```
не хватает округления картинок

mmbabaev commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,10 +1,173 @@
+            correctAnswers += 1
+        }
+        
+        imageView.layer.masksToBounds = true
```
также нужно выставить contentMode = aspectFill

mmbabaev commented:
не проставляются нужные шрифты. Заметили ошибку в учебнике, исправим. Нужно тут поставить галочки в таргете
<img width="1368" alt="Screen Shot 2022-11-17 at 00 30 03" src="https://user-images.githubusercontent.com/12600807/202298599-ab0d27f3-bcef-4f62-b236-232b851a0436.png">

mmbabaev commented:
шрифты не меняются в приложении. В учебнике допущена ошибка, поэтому этот пункт необязателен. 
На скрине показал как нужно сделать: 
1) Шрифты добавлены в Group (серая папка)
2) Шрифтам проставлена галочка Target Membership  (справа)

<img width="1356" alt="Screen Shot 2022-11-17 at 20 15 01" src="https://user-images.githubusercontent.com/12600807/202513128-d74c5681-7a76-4778-ae5b-0fbcd44f0893.png">

Полезная статья, которая помогла мне разобраться в свое время: 
https://codewithchris.com/common-mistakes-with-adding-custom-fonts-to-your-ios-app/

mmbabaev commented:
Не хватает bottom constraint от корневой StackView к SafeArea. После этого проверьте интерфейс на iPhone 13 mini

mmbabaev commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,10 +1,173 @@
+        imageView.layer.borderWidth = 8
+        imageView.layer.borderColor = isCorrect ? UIColor.ypGreen.cgColor : UIColor.ypRed.cgColor
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
```
Необязательно: если быстро нажимать одну и ту же кнопку, то срабатывают лишние нажатия, пока мы ожидаем следующего вопроса. Чтобы этого не было, можно использовать свойство isEnabled. При начале ожидания "выключать" кнопку, после ожидания "включать" https://developer.apple.com/documentation/uikit/uicontrol/1618217-isenabled

etdevios committed:
ifx:
- шрифты добавлены в Group, шрифтам проставлена галочка Target.
- добавлен конкстрейнтдля корневой StackView
- добавлено округление imageView
- изменено наполняемость для imageView
- реализовано ожидание для кнопок

mmbabaev commented:
Все замечания исправлены, но теперь сжимается label на iphone 13 mini 
<img width="452" alt="Screen Shot 2022-11-20 at 23 20 48" src="https://user-images.githubusercontent.com/12600807/202924326-0b7d31de-9cef-4d20-aa21-648bd431030a.png">

etdevios committed:
fix: изменен Autoshrink в Question Label на MinimumFontScale для корректного отображения текста на iPhone 13 mini

mmbabaev commented:
Интересное решение, принял. Вообще еще можно констрейнты top и bottom здесь уменьшить
<img width="301" alt="Screen Shot 2022-11-21 at 16 32 38" src="https://user-images.githubusercontent.com/12600807/203068417-d3b524b3-66f3-4b81-9b70-b516f507d530.png">

ABashkirova commented:
Можно вмержить
